himalaya/backend/torch_cuda.py

```diff
@@ -2,15 +2,22 @@
 
 from .torch import *  # noqa
 import torch
+import sys
 
 try:
     torch.arange(1).cuda()
 except AssertionError as error:
-    import sys
     if "pytest" in sys.modules:  # if run through pytest
         import pytest
         pytest.skip("PyTorch not compiled with CUDA enabled.")
     raise AssertionError("PyTorch not compiled with CUDA enabled.") from error
+except RuntimeError as error:
```

Comment:
I don't think there is value in having two different messages (one about cuda, one about nvidia). In fact, I don't remember why I catch an AssertionError here, and the try/catch is just hanging on my laptop when I call `set_backend("torch_cuda")`.

I think we should change the logic, using `torch.cuda.is_available()` instead of a try/catch clause, and change the AssertionError into a RuntimeError. I don't have a preference for the message.

Reply:
sounds good, I'll try to change that and see what the tests look like